log/field_encoders.go

```diff
@@ -0,0 +1,49 @@
+package log
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+)
+
+// FieldEncoder describes a way to encode fields.
+type FieldEncoder func(io.Writer, Field) error
```

Comment:
I don't really understand why you would want to encode 1 field at a time, instead of always working with many fields (and 1 field being a special case of many-fields).